examples/modx: Make the modx example work with the Cortex M0

aykevl opened the pull request:
This was needed to get it to work on a Cortex M0 (nRF51822).
I've also made the Makefile a bit more flexible.

e.g.

    make CROSS=arm-cortex-m0

aykevl committed:
examples/modx: Make the modx example work with the Cortex M0

dpgeorge commented:
Since one will always be "cross compiling" for some target architecture I think it makes more sense to call the variable ARCH and make the valid values match the MP_PERSISTENT_ARCH constants.  Eg: `make ARCH=thumb2`.  And then it would always use the $ARCH variable to make the build directory's name.

Also, I think M0 and M4 should have separate MP_PERSISTENT_ARCH constants, because M4 code won't run on M0.

aykevl commented:
I've updated the pull request.

I'm not so sure about the naming. I've changed thumb2 to m0 and m4. But the architecture itself is something else - ARM 6M or ARM 7EM (or ARM 7M for the Cortex M3), so it could also be e.g. arm-6m for the M0. But using M0 etc. seems more obvious to me.

Also, maybe M4 is the wrong name here - it includes floating point support so it should maybe be called M4F (see: https://en.wikipedia.org/wiki/ARM_Cortex-M#Instruction_sets).

aykevl committed:
examples/modx: Ignore build directories.

dpgeorge commented:
Thanks for updating.

I also agree that the thumb naming needs a bit more refining.  Note there is also M7 with extra instructions (like double precision fp) so at least that needs to be added.  But all this can be done later.  Merged in b7efae3b2daf84809ee24451590fb4f1920a2123
